test/unit/redirect.test.js

```diff
@@ -0,0 +1,211 @@
+/**
```

Comment:
need to add tests for redirects to the same host:
* both are region1.cloud.ibm.com
* both are region2.notcloud.ibm.com